packages/components/src/components/select/select/select.tsx

```diff
@@ -344,6 +354,7 @@ export class Select {
     }
     this.updateMenuState(false);
     this.combobox.focus();
+    this.hasSlottedImage = this.getSelectedOptionImagePath(this.selectOptions) !== '';
```

Comment:
Maybe it would be safer to update this in the value watcher? Test in E2E if dynamic value change would update the image